pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authentication/AuthenticationProviderToken.java

```diff
@@ -375,6 +383,7 @@ public String getAuthRole() throws AuthenticationException {
         public AuthData authenticate(AuthData authData) throws AuthenticationException {
             String token = new String(authData.getBytes(), UTF_8);
             checkExpiration(token);
+            this.authenticationDataSource = new AuthenticationDataCommand(token, remoteAddress, sslSession);
```

Comment:
Is there any thread safety concerns here? would it make sense to make authenticationDataSource field `volatile`?